frontend/src/components/pages/agents/details/a2a/chat/components/chat-input.tsx

```diff
@@ -75,20 +112,35 @@ export const ChatInput = ({
         </PromptInputBody>
         <PromptInputFooter>
           <PromptInputTools>
-            {model && (
-              <PromptInputModelSelect disabled value={model}>
+            {agent.model && (
+              <PromptInputModelSelect disabled value={agent.model}>
                 <PromptInputModelSelectTrigger>
                   <PromptInputModelSelectValue>
-                    <AIAgentModel model={model} size="sm" />
+                    <AIAgentModel model={agent.model} size="sm" />
                   </PromptInputModelSelectValue>
                 </PromptInputModelSelectTrigger>
                 <PromptInputModelSelectContent>
-                  <PromptInputModelSelectItem value={model}>
-                    <AIAgentModel model={model} size="sm" />
+                  <PromptInputModelSelectItem value={agent.model}>
+                    <AIAgentModel model={agent.model} size="sm" />
                   </PromptInputModelSelectItem>
                 </PromptInputModelSelectContent>
               </PromptInputModelSelect>
             )}
+            <Context
+              maxTokens={usage.max_input_tokens || 272_000}
```

Comment:
272k is only for some models, should we just not show the context at all when `max_input_tokens === undefined`